x-pack/filebeat/input/httpjson/request.go

```diff
@@ -57,6 +63,44 @@ func (c *httpClient) do(stdCtx context.Context, req *http.Request) (*http.Respon
 	return resp, nil
 }
 
+// tracer is a lightweight transaction tracer for debugging httpjson configurations.
+// It is not intended to be used for machine ingestion, but rather as a human aid.
+type tracer struct {
+	sess int64 // sess is the unix time of the start of a session.
+	txn  int   // txn is a session-unique transaction id.
+	log  *log.Logger
+}
+
+func newTracer(logger *lumberjack.Logger) *tracer {
+	if logger == nil {
+		return nil
+	}
+	return &tracer{sess: time.Now().UnixMilli(), log: log.New(logger, "", log.LstdFlags)}
+}
+
+func (t *tracer) trace(req *http.Request, resp *http.Response) {
+	if t == nil {
+		return
+	}
+	t.txn++
+	b, err := httputil.DumpRequestOut(req, true)
+	if err != nil {
+		t.log.Print(err)
+	} else {
+		t.log.Printf("request %d.%d:\n\t%s", t.sess, t.txn, bytes.ReplaceAll(b, []byte{'\n'}, []byte("\n\t")))
```

Comment:
[Question]
Why modify the original data? Why add this tab character at the beginning of new lines?

If the goal is to help debug issues, I believe it's better to keep the original data untouched.

Reply:
The log file is very hard to follow if it is not indented. This just moves the req and resp in one tab, essentially as the testing output does.

Reply:
I get it now! Then it makes sense :smile: